OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -85,32 +86,30 @@ public static Func<int> TurretFacingFromInit(IActorInitializer init, int def, st
 		}
 
 		public Turreted(ActorInitializer init, TurretedInfo info)
+			: base(info)
 		{
 			this.info = info;
 			TurretFacing = TurretFacingFromInit(init, info.InitialFacing, info.Turret)();
 		}
 
 		public void Created(Actor self)
 		{
-			attack = self.TraitOrDefault<AttackTurreted>();
+			attack = self.TraitsImplementing<AttackTurreted>().ToArray();
 			facing = self.TraitOrDefault<IFacing>();
 			body = self.Trait<BodyOrientation>();
 		}
 
 		public virtual void Tick(Actor self)
 		{
 			// NOTE: FaceTarget is called in AttackTurreted.CanAttack if the turret has a target.
-			if (attack != null)
+			if (!attack.Any(a => a.IsAttacking && !a.IsTraitDisabled))
```

Comment:
If we don't have any `AttackTurreted` traits this condition will be true (although it shouldn't).